repository/src/main/java/org/alfresco/repo/action/executer/NodeSizeDetailActionExecutor.java

```diff
@@ -0,0 +1,213 @@
+                List<Pair<String, Integer>> pairSizes = results.getFieldFacet(FIELD_FACET).subList(skipCount, totalItems);
+                pairSizes.parallelStream().forEach(id -> {
+                    try
+                    {
+                        if(id.getSecond()>0)
+                        {
+                            resultSize.addAndGet(Long.valueOf(id.getFirst()) * id.getSecond());
+                        }
+                    }
+                    catch (Exception e)
+                    {
+                        resultSize.addAndGet(0);
+                    }
+                });
+
+                totalSizeFromFacet+=resultSize.longValue();
+                resultSize.set(0);
+
+                if (results.getFieldFacet(FIELD_FACET).size() <= totalItems || results.getFieldFacet(FIELD_FACET).size() <= defaultItems)
+                {
+                    isCalculationCompleted = true;
+                }
+                else
+                {
+                    skipCount += defaultItems;
+                    int remainingItems = results.getFieldFacet(FIELD_FACET).size() - totalItems;
+                    totalItems += Math.min(remainingItems, defaultItems);
+                }
+            }
+        }
+        catch (RuntimeException runtimeException)
+        {
+            LOG.error("Exception occurred in NodeSizeDetailActionExecutor:results {} ", runtimeException.getMessage());
+            response.put(EXCEPTION,"Exception occurred in NodeSizeDetailActionExecutor:results {} "+runtimeException.getMessage());
+            simpleCache.put(nodeRef.getId(),response);
+            throw runtimeException;
+        }
+
+        LOG.debug(" Calculating size of Folder Node - NodeSizeDetailActionExecutor:executeImpl ");
+        final LocalDateTime eventTimestamp = LocalDateTime.ofInstant(Instant.now(), ZoneId.systemDefault());
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy:MM:dd HH:mm:ss");
+        String formattedTimestamp = eventTimestamp.format(formatter);
+        response.put("nodeId", nodeRef.getId());
+        response.put("size", totalSizeFromFacet);
+        response.put("calculatedAt", formattedTimestamp);
+        response.put("numberOfFiles", results != null ? results.getNodeRefs().size() : 0);
+        response.put("actionId", nodeAction.getId());
+
+        if(isCalculationCompleted)
+        {
+            simpleCache.put(nodeRef.getId(),response);
+        }
+    }
+
+    protected ResultSet facetQuery(NodeRef nodeRef)
+    {
+        StringBuilder aftsQuery = new StringBuilder();
+        aftsQuery.append("ANCESTOR:\"").append(nodeRef).append("\" AND TYPE:content");
+        String query = aftsQuery.toString();
+
+        SearchParameters searchParameters = new SearchParameters();
+        searchParameters.addStore(StoreRef.STORE_REF_WORKSPACE_SPACESSTORE);
+        searchParameters.setLanguage(SearchService.LANGUAGE_FTS_ALFRESCO);
+        searchParameters.setQuery(query);
```

Comment:
Elasticsearch's default total hit limit is set to 10,000. Therefore, for a folder containing more than 10,000 items, the result will not be accurate. You need to override this default to allow for unlimited hits to ensure completeness of the results.

Reply:
Updated
Done